compiler/rustc_resolve/src/late.rs

```diff
@@ -4810,6 +4806,17 @@ impl<'a, 'ast, 'ra: 'ast, 'tcx> LateResolutionVisitor<'a, 'ast, 'ra, 'tcx> {
         res
     }
 
+    fn is_invalid_proc_macro_item_for_doc(&self, did: DefId) -> bool {
+        if !self.r.tcx.crate_types().contains(&CrateType::ProcMacro)
+            || !matches!(self.r.tcx.sess.opts.resolve_doc_links, ResolveDocLinks::ExportedMetadata)
```

Comment:
Maybe put the resolve_doc_links check first. It is probably a bit cheaper than checking the crate type.